code/procs/jobprocs.dm

```diff
@@ -547,18 +547,45 @@
 
 	return
 
-/// Equip items from sensory traits
-/mob/living/carbon/human/proc/equip_sensory_items()
+/// Equip items from traits and put items that have been over written into hands/pockets
+/mob/living/carbon/human/proc/equip_trait_items() //could probably use varibles in the proc to determine what needs to be equipped and in what slot but im lazy as hell
 	if (src.traitHolder.hasTrait("blind"))
-		src.drop_from_slot(src.glasses)
+		src.store_from_slot(src.glasses)
 		src.equip_if_possible(new /obj/item/clothing/glasses/visor(src), src.slot_glasses)
 	if (src.traitHolder.hasTrait("shortsighted"))
-		src.drop_from_slot(src.glasses)
+		src.store_from_slot(src.glasses)
 		src.equip_if_possible(new /obj/item/clothing/glasses/regular(src), src.slot_glasses)
 	if (src.traitHolder.hasTrait("deaf"))
-		src.drop_from_slot(src.ears)
+		src.store_from_slot(src.ears)
 		src.equip_if_possible(new /obj/item/device/radio/headset/deaf(src), src.slot_ears)
 
+/mob/living/carbon/human/proc/store_from_slot(obj/item/item,) //this is in here over mob.dm because equip_if_possible wouldnt work lmao
```

Comment:
```suggestion
/mob/living/carbon/human/proc/store_from_slot(obj/item/item) //this is in here over mob.dm because equip_if_possible wouldnt work lmao
```
It should probably still be in `mob.dm` I think?